app/src/androidTest/java/ch/hikemate/app/ui/map/HikeDetailsScreenTest.kt

```diff
@@ -508,4 +567,123 @@ class HikeDetailScreenTest {
+    }
+
+    verify(facilitiesRepository).getFacilities(any(), any(), any())
+  }
+
+  @Test
+  fun hikeDetails_displaysCorrectDrawableForFacilityType() = runBlocking {
+    // Setup a detailed hike
+    setUpSelectedHike(detailedHike2)
+
+    val bounds = detailedHike2.bounds.toBoundingBox()
+    val center = LatLong(bounds.centerLatitude, bounds.centerLongitude)
+    val testFacility = Facility(type = FacilityType.TOILETS, coordinates = center)
+
+    val facilities: MutableState<List<Facility>?> = mutableStateOf(listOf(testFacility))
+
+    lateinit var mapView: MapView
+    lateinit var context: Context
+
+    composeTestRule.setContent {
+      context = LocalContext.current
+      mapView = hikeDetailsMap(detailedHike2, facilitiesViewModel, facilities)
+    }
+    composeTestRule.waitForIdle()
+
+    // Create a custom waiter that checks for the marker presence
+    var attempts = 0
+    val maxAttempts = 500 // Adjust as needed
+    val delayMs = 100L // Small delay between checks
+
+    // Wait for the marker to appear using polling
+    while (attempts < maxAttempts) {
+      val facilityMarkers =
+          mapView.overlays.filterIsInstance<Marker>().filter {
+            it.relatedObject == R.string.facility_marker
+          }
+
+      if (facilityMarkers.isNotEmpty()) {
+        // Marker found, proceed with assertions
+        val marker = facilityMarkers.first()
+        val expectedDrawable = ContextCompat.getDrawable(context, R.drawable.toilets)
+
+        assertEquals(1, facilityMarkers.size)
+        assertTrue(
+            "Marker should have correct drawable icon",
+            areSameDrawable(expectedDrawable, marker.icon))
+        assertEquals(testFacility.coordinates.lat, marker.position.latitude, 0.0001)
+        assertEquals(testFacility.coordinates.lon, marker.position.longitude, 0.0001)
+        return@runBlocking // Exit successfully
+      }
+
+      delay(delayMs) // Use coroutine delay instead of Thread.sleep
+      attempts++
+    }
+
+    // If we get here, the marker never appeared
+    fail("Marker was not added to map after ${maxAttempts * delayMs}ms")
+  }
+
+  @Test
+  fun hikeDetails_hidesFacilities_whenZoomLevelIsInsufficient() = runTest {
+    setUpSelectedHike(detailedHike3)
+
+    val testFacilities =
+        listOf(Facility(type = FacilityType.TOILETS, coordinates = LatLong(45.9, 7.6)))
+
+    val facilities: MutableState<List<Facility>?> = mutableStateOf(testFacilities)
+    lateinit var mapView: MapView
+    val minZoomForFacilities = MIN_ZOOM_FOR_FACILITIES
+
+    composeTestRule.setContent {
+      mapView = hikeDetailsMap(detailedHike3, facilitiesViewModel, facilities)
+    }
+
+    composeTestRule.waitForIdle()
+    Thread.sleep(3000)
+
+    // Verify facilities are hidden at insufficient zoom levels
+    val finalMarkers =
+        mapView.overlays.filterIsInstance<Marker>().filter {
+          it.relatedObject == R.string.facility_marker
+        }
+
+    assertTrue(
```

Comment:
Question: Might be a dumb question, but how come the zoom level is actually to far zoomed out in this test? Is it that the hike is "big" enough for this effect to happen?

Reply:
The bounds are too big.